### PR TITLE
expression: implement vectorized evaluation for `builtinExportSet*ArgSig`

### PR DESCRIPTION
**Pull request by mmyj**
PCP #12106

### What problem does this PR solve? 
implement vectorized evaluation for `builtinExportSet*ArgSig`, for #12106

### What is changed and how it works?
```
goos: linux
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinStringFunc/builtinExportSet3ArgSig-VecBuiltinFunc-3                100          11189283 ns/op         9208039 B/op     129024 allocs/op
BenchmarkVectorizedBuiltinStringFunc/builtinExportSet3ArgSig-NonVecBuiltinFunc-3              99          11763305 ns/op         9207929 B/op     129024 allocs/op
BenchmarkVectorizedBuiltinStringFunc/builtinExportSet4ArgSig-VecBuiltinFunc-3                100          10818161 ns/op         9208019 B/op     129024 allocs/op
BenchmarkVectorizedBuiltinStringFunc/builtinExportSet4ArgSig-NonVecBuiltinFunc-3             100          10978643 ns/op         9207918 B/op     129024 allocs/op
BenchmarkVectorizedBuiltinStringFunc/builtinExportSet5ArgSig-VecBuiltinFunc-3                967           1246358 ns/op          262147 B/op      18432 allocs/op
BenchmarkVectorizedBuiltinStringFunc/builtinExportSet5ArgSig-NonVecBuiltinFunc-3             866           1344990 ns/op          262147 B/op      18432 allocs/op
PASS
ok      github.com/pingcap/tidb/expression      7.234s
```

### Check List 

Tests 

- Unit test

### CODE REVIEW

**Comment by XuHuaiyu** [comment]
@mmyj 
Kindly reminded, you will get 150 points if you split this PR into 3 PRs.

File: expression/builtin_string_vec.go [hunk 1]
```diff
@@ -1119,11 +1201,66 @@ func (b *builtinInsertSig) vecEvalString(input *chunk.Chunk, result *chunk.Colum
...
+		return err
+	}
+	defer b.bufAllocator.put(numberOfBits)
+	if err := b.args[4].VecEvalString(b.ctx, input, numberOfBits); err != nil {
+		return err
+	}
+	result.ReserveString(n)
+	bits.MergeNulls(numberOfBits)
+	i64s := bits.Int64s()
+	i64s2 := numberOfBits.Int64s()
```
Comment (XuHuaiyu) [inline expression/builtin_string_vec.go]: Should we check this? 	
``` go
if numberOfBits < 0 || numberOfBits > 64 {
		numberOfBits = 64
}
```

Comment (mmyj) [reply expression/builtin_string_vec.go]: I will fix it.

File: expression/builtin_string_vec_test.go [hunk 2]
```diff
@@ -162,7 +162,23 @@ var vecBuiltinStringCases = map[string][]vecExprBenchCase{
...
+		},
+		{
+			retEvalType:   types.ETString,
+			childrenTypes: []types.EvalType{types.ETInt, types.ETString, types.ETString, types.ETString},
+			geners:        []dataGenerator{&rangeInt64Gener{10, 100}, &constStrGener{"Y"}, &constStrGener{"N"}, &constStrGener{","}},
+		},
+		{
+			retEvalType:   types.ETString,
+			childrenTypes: []types.EvalType{types.ETInt, types.ETString, types.ETString, types.ETString, types.ETInt},
+			geners:        []dataGenerator{&rangeInt64Gener{10, 100}, &constStrGener{"Y"}, &constStrGener{"N"}, &constStrGener{","}, &rangeInt64Gener{10, 11}},
```
Comment (XuHuaiyu) [inline expression/builtin_string_vec_test.go]: Use `rangeInt64Gener{-10, 70}` for the last argument.

Comment (mmyj) [reply expression/builtin_string_vec_test.go]: I will fix it.

File: expression/builtin_string_vec.go [hunk 3]
```diff
@@ -1261,11 +1343,69 @@ func (b *builtinInsertSig) vecEvalString(input *chunk.Chunk, result *chunk.Colum
...
+	defer b.bufAllocator.put(separator)
+	if err := b.args[3].VecEvalString(b.ctx, input, separator); err != nil {
+		return err
+	}
+	numberOfBits, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(numberOfBits)
+	if err := b.args[4].VecEvalString(b.ctx, input, numberOfBits); err != nil {
```
Comment (Reminiscent) [inline expression/builtin_string_vec.go]: Here should be ```err := b.args[4].VecEvalInt(b.ctx, input, numberOfBits)```

**Pull request merged**